### PR TITLE
CI: add a docker image

### PR DESCRIPTION
**Pull request by bilelmoussaoui**
This adds a `Dockerfile` and a CI job that builds the image and publish it on Github Packages containers registry. A future PR will migrate the CI to use it as we need to merge it first to get the image built & published.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:20.10
+
+RUN apt update -y
+RUN apt install -y libgtk-3-dev \
```
Comment (GuillaumeGomez) [inline Dockerfile:4]: Please put `libgtk-3-dev` on a new line.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    libglib2.0-dev \
+    libgraphene-1.0-dev \
+    git xvfb curl \
```
Comment (GuillaumeGomez) [inline Dockerfile:7]: Please put all packages on their own line.

**Review by GuillaumeGomez — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
Thanks! Merging then. :)

**Comment by MarijnS95** [comment]
@bilelmoussaoui The push-action needs to check if the parent organization is `gtk-rs` instead of trying to push it to forks: https://github.com/MarijnS95/gtk3-rs/runs/3454094148?check_suite_focus=true

**Comment by bilelmoussaoui** [comment]
Feel free to submit a patch, normally the job shouldn't be triggered by someone's fork unless they update the Dockerfile

**Comment by MarijnS95** [comment]
Ah yes I rebased and pushed a branch that hasn't been touched in 3 months, that probably caused the change :)

**Pull request merged**